src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthModule.TransactionExecutor.cs

```diff
@@ -0,0 +1,180 @@
+                }
+
+                transactionCall.From ??= Address.SystemUser;
+            }
+        }
+
+        private class CallTransactionExecutor : TransactionExecutor<string>
+        {
+            public CallTransactionExecutor(IBlockchainBridge blockchainBridge, IBlockFinder blockFinder, TimeSpan cancellationTokenTimeout, IJsonRpcConfig rpcConfig)
+                : base(blockchainBridge, blockFinder, cancellationTokenTimeout, rpcConfig)
+            {
+            }
+
+            protected override ResultWrapper<string> GetExecutionError(Transaction tx, BlockchainBridge.CallOutput result) => 
+                ResultWrapper<string>.Fail("VM execution error.", ErrorCodes.ExecutionError, result.Error);
+
+            protected override string GetResult(Transaction tx, BlockchainBridge.CallOutput result) => 
+                result.OutputData.ToHexString(true);
+
+            protected override BlockchainBridge.CallOutput ExecuteTx(BlockHeader header, Transaction tx, CancellationToken token) => 
+                _blockchainBridge.Call(header, tx, token);
+        }
+
+        private class EstimateGasTransactionExecutor : TransactionExecutor<UInt256?>
+        {
+            public EstimateGasTransactionExecutor(IBlockchainBridge blockchainBridge, IBlockFinder blockFinder, TimeSpan cancellationTokenTimeout, IJsonRpcConfig rpcConfig) 
+                : base(blockchainBridge, blockFinder, cancellationTokenTimeout, rpcConfig)
+            {
+            }
+
+            protected override BlockchainBridge.CallOutput ExecuteTx(BlockHeader header, Transaction tx, CancellationToken token) => 
+                _blockchainBridge.EstimateGas(header, tx, token);
+
+            protected override UInt256? GetResult(Transaction tx, BlockchainBridge.CallOutput result) => (UInt256)result.GasSpent;
+
+            protected override ResultWrapper<UInt256?> GetExecutionError(Transaction tx, BlockchainBridge.CallOutput result) => 
+                ResultWrapper<UInt256?>.Fail(result.Error, ErrorCodes.InternalError);
+        }
+        
+        private class CreateAccessListTransactionExecutor : TransactionExecutor<AccessListForRpc>
+        {
+            private readonly bool _optimize;
+
+            public CreateAccessListTransactionExecutor(IBlockchainBridge blockchainBridge, IBlockFinder blockFinder, TimeSpan cancellationTokenTimeout, IJsonRpcConfig rpcConfig, bool optimize) 
+                : base(blockchainBridge, blockFinder, cancellationTokenTimeout, rpcConfig)
+            {
+                _optimize = optimize;
+            }
+
+            protected override BlockchainBridge.CallOutput ExecuteTx(BlockHeader header, Transaction tx, CancellationToken token) => 
+                _blockchainBridge.CreateAccessList(header, tx, token, _optimize);
+
+            protected override AccessListForRpc GetResult(Transaction tx, BlockchainBridge.CallOutput result) =>
```

Comment:
simplify by embedding in ExecuteTx
much complexity has been added through the 3 small abstract methods to avoid repeating 2 simple lines